Partial Sync SVGTextContentElement.idl with IDL Spec

Ahmad-S792 opened the pull request:
#### 7a9711ddf39ee9a4428602413fc2f4beecbf320c
<pre>
Partial Sync SVGTextContentElement.idl with IDL Spec

<a href="https://bugs.webkit.org/show_bug.cgi?id=120451">https://bugs.webkit.org/show_bug.cgi?id=120451</a>

Reviewed by Said Abou-Hallawa.

This patch aligns WebKit with web-spec [1].

[1] <a href="https://svgwg.org/svg2-draft/text.html#InterfaceSVGTextContentElement">https://svgwg.org/svg2-draft/text.html#InterfaceSVGTextContentElement</a>

Further, add FIXME to fully align with web-spec.

* Source/WebCore/svg/SVGTextContentElement.idl:
* LayoutTests/imported/w3c/web-platform-tests/svg/idlharness.window-expected.txt: Rebaselined

Canonical link: <a href="https://commits.webkit.org/262964@main">https://commits.webkit.org/262964@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/644920b0b85a41b9602f6046ab75c0bed7492bcf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3169 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3230 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3340 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4582 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3527 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/3137 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3294 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3257 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/2757 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3200 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3519 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2850 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4393 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1017 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2823 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/2665 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2796 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2874 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/4140 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3245 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/2591 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2829 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2825 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/768 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2822 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3088 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

shallawa commented on Source/WebCore/svg/SVGTextContentElement.idl:
```diff
@@ -31,16 +33,17 @@
+    // FIXME: SVGPoint/SVGRect should be DOMPoint/DOMRect as per web-spec.
+    [NewObject] SVGPoint getStartPositionOfChar(unsigned long charnum);
+    [NewObject] SVGPoint getEndPositionOfChar(unsigned long charnum);
+    [NewObject] SVGRect getExtentOfChar(unsigned long charnum);
```
Maybe we need to remove `[NewObject]` since it is the default. I think the specs rule is to add `[SameObject]` if the return object is an attribute of the owner. Otherwise it is just a calculated value.

Ahmad-S792 replied:
Done. Removed [NewObject].

Ahmad-S792 committed:
Partial Sync SVGTextContentElement.idl with IDL Spec

https://bugs.webkit.org/show_bug.cgi?id=120451

Reviewed by Said Abou-Hallawa.

This patch aligns WebKit with web-spec [1].

[1] https://svgwg.org/svg2-draft/text.html#InterfaceSVGTextContentElement

Further, add FIXME to fully align with web-spec.

* Source/WebCore/svg/SVGTextContentElement.idl:
* LayoutTests/imported/w3c/web-platform-tests/svg/idlharness.window-expected.txt: Rebaselined

Canonical link: https://commits.webkit.org/262964@main
